### PR TITLE
Add more tests for serializers

### PR DESCRIPTION
**Pull request by leplatrem**
@tarekziade @n1k0 r?

### CODE REVIEW

File: kinto_signer/tests/test_serializer.py [hunk 1]
```diff
@@ -86,3 +148,23 @@ def test_canonical_json_with_deeply_nested_dicts():
...
+# XXX: those will probably require to rewrite the whole suite of tests.
+#
+# def test_serializes_nan_to_null():
+#     records = [{'id': '4', 'a': float('nan')}]
+#     assert canonical_json(records) == '[{"a":null,"id":4}]'
+#
+#
+# def test_serializes_numbers_as_is():
+#     records = [{'id': '4', 'a': -0, 'b': 3.50}]
+#     assert canonical_json(records) == '[{"a":-0,"b":3.50,"id":"4"}]'
```
Comment (leplatrem) [inline kinto_signer/tests/test_serializer.py]: This one seems pretty hard to obtain :(

**Comment by n1k0** [comment]
Looks good but I'm puzzled with the commented tests indeed.

**Comment by n1k0** [comment]
Seems like there's at least a way to define our own JSON encoder https://docs.python.org/2/library/json.html#json.JSONEncoder

File: kinto_signer/tests/test_serializer.py (hunk 1 above)
Comment (Natim) [inline kinto_signer/tests/test_serializer.py]: ``-0`` doesn't seems to be something we want do we?

Comment (almet) [inline kinto_signer/tests/test_serializer.py]: In python, it doesn't seem possible to serialize `-0` and `0` differently. You can with `-0.0` and `+0.0` though. 

See http://bugs.python.org/msg59137 and http://bugs.python.org/msg59154

**Comment by almet** [comment]
The idea to serialize -0 and +0 differently came form [the IETF draft from rundgren](http://webpki.org/ietf/draft-rundgren-predictable-serialization-for-json-tools-00.html).

Given the fact that it's not easy to do with Python (and not reliable across languages), we might want to drop support for such corner case?

File: kinto_signer/tests/test_serializer.py [hunk 2]
```diff
@@ -86,3 +148,23 @@ def test_canonical_json_with_deeply_nested_dicts():
...
     assert canonical_json(records) == expected
+
+
+#
+# Failing tests
+#
+# XXX: those will probably require to rewrite the whole suite of tests.
+#
+# def test_serializes_nan_to_null():
+#     records = [{'id': '4', 'a': float('nan')}]
```
Comment (almet) [inline kinto_signer/tests/test_serializer.py]: I don't think Python got a concept of "not a number / nan" so we probably should just throw this test.

File: kinto_signer/tests/test_serializer.py [hunk 3]
```diff
@@ -86,3 +148,23 @@ def test_canonical_json_with_deeply_nested_dicts():
...
+#     assert canonical_json(records) == '[{"a":null,"id":4}]'
+#
+#
+# def test_serializes_numbers_as_is():
+#     records = [{'id': '4', 'a': -0, 'b': 3.50}]
+#     assert canonical_json(records) == '[{"a":-0,"b":3.50,"id":"4"}]'
+#
+#
+# def test_serializes_infinity_to_null():
+#     records = [{'id': '4', 'a': float("inf")}]
```
Comment (almet) [inline kinto_signer/tests/test_serializer.py]: In python, some pointers: http://stackoverflow.com/a/7781273
